modules/plugin/geotiff/src/test/java/org/geotools/gce/geotiff/GeoTiffReaderCogOnlineTest.java

```diff
@@ -61,4 +75,49 @@ public void testCogRead() throws URISyntaxException, IOException {
         assertEquals(512, raster.getHeight());
         assertEquals(1, raster.getNumBands());
     }
+
+    @Test
+    public void testCogOverview() throws URISyntaxException, IOException {
+        GeoTiffReader reader = new GeoTiffReader(getInputProvider());
+        assertEquals(
+                "LC08_L1TP_153075_20190515_20190515_01_RT_B2", reader.getGridCoverageNames()[0]);
+
+        GeneralParameterValue[] params = new GeneralParameterValue[1];
+        // Define a GridGeometry in order to reduce the output
+        final ParameterValue<GridGeometry2D> gg =
+                AbstractGridFormat.READ_GRIDGEOMETRY2D.createValue();
+        final GeneralEnvelope envelope = reader.getOriginalEnvelope();
+        final Dimension dim = new Dimension();
+        dim.setSize(
+                reader.getOriginalGridRange().getSpan(0) / 24,
+                reader.getOriginalGridRange().getSpan(1) / 24);
+        final Rectangle rasterArea = ((GridEnvelope2D) reader.getOriginalGridRange());
+        rasterArea.setSize(dim);
+        final GridEnvelope2D range = new GridEnvelope2D(rasterArea);
+        gg.setValue(new GridGeometry2D(range, envelope));
+        params[0] = gg;
+
+        GridCoverage2D coverage = reader.read(params);
+        assertNotNull(coverage);
+        RenderedImage image = coverage.getRenderedImage();
+        int numTileX = image.getNumXTiles();
+        int numTileY = image.getNumYTiles();
+
+        CogImageReader imageReader = (CogImageReader) image.getProperty("JAI.ImageReader");
```

Comment:
Interesting bit... made me wonder about a few things that are related to COG, but not the ovr usage per se.

1) The coverage generated by image mosaic normally contains a list of files used to generated it. Is that still the case for COG? 
But I guess it's the main file, not the ovr, that is listed there... still useful I believe.

2) Also wondering how CSW direct download would play with COGs.

Reply:
1. Updated to deal with that, reporting the COG Url. Note that ImageMosaic reports .ovr instead, so it will be the same for COG

2. All GT APIs used by DirectDownload are File-related (FileGroups, FileGroupProvider, FileInfo and so on). We need to revisit the GT API to have it working for cog too